packages/neon/neon_files/lib/blocs/files.dart

```diff
@@ -115,36 +115,36 @@ class FilesBloc extends InteractiveBloc implements FilesBlocEvents, FilesBlocSta
   }
 
   @override
-  void removeFavorite(final List<String> path) {
+  void removeFavorite(final PathUri path) {
     wrapAction(
       () async => account.client.webdav.proppatch(
-        Uri(pathSegments: path),
+        path,
         set: WebDavProp(ocfavorite: 0),
       ),
     );
   }
 
   @override
-  void rename(final List<String> path, final String name) {
+  void rename(final PathUri path, final String name) {
     wrapAction(
       () async => account.client.webdav.move(
-        Uri(pathSegments: path),
-        Uri(pathSegments: List.from(path)..last = name),
+        path,
+        PathUri(pathSegments: List.from(path.pathSegments)..last = name),
       ),
     );
   }
 
   @override
-  void syncFile(final List<String> path) {
+  void syncFile(final PathUri path) {
     wrapAction(
       () async {
         final file = File(
-          p.join(
+          p.joinAll([
             await NeonPlatform.instance.userAccessibleAppDataPath,
             account.humanReadableID,
             'files',
```

Comment:
Not directly related so we can open an issue but we should have a convenience method in the `NeonPlatform` handling this for us.

Just give it the current AppID and get a path to use. This will become especially handy with the sync work.